fdbserver/DataDistribution.actor.cpp

```diff
@@ -1425,8 +1425,11 @@ ACTOR Future<Void> ddSnapCreateCore(DistributorSnapRequest snapReq, Reference<As
 				coordSnapReqs.push_back(trySendSnapReq(
 				    interf.workerSnapReq, WorkerSnapRequest(snapReq.snapPayload, snapReq.snapUID, "coord"_sr)));
 		}
-		auto const coordFaultTolerance = std::min<int>(std::max<int>(0, coordSnapReqs.size() / 2 - 1),
-		                                               SERVER_KNOBS->MAX_COORDINATOR_SNAPSHOT_FAULT_TOLERANCE);
+		// At present, the fault injection workload doesn't repect the KNOB MAX_COORDINATOR_SNAPSHOT_FAULT_TOLERANCE
+		// Consequently, we ignore it in simulation tests
+		auto const coordFaultTolerance = std::min<int>(
+		    std::max<int>(0, (coordSnapReqs.size() - 1) / 2),
+		    g_simulator->isSimulated() ? coordSnapReqs.size() : SERVER_KNOBS->MAX_COORDINATOR_SNAPSHOT_FAULT_TOLERANCE);
```

Comment:
Since fault injection doesn't acknowledge the fact that some coordinators can not be snapshotted, shouldn't `coordFaultTolerance` be 0 in simulation?

Reply:
I may not fully follow the question here.
But coordinators can be killed due to fault injection and remain unavailable while taking snapshots.
E.g., 2 out of 5 coordinators were killed, and the knob is making the fault tolerance still 1, but actually, we can tolerate 2 failed coordinators here. So it just ignores the knob in the simulation.

Reply:
Oh I see, I thought the problem was on restore (that we couldn't even tolerate one coordinator down due to attrition). But actually it sounds like the problem is when taking a snapshot. In that case, this change makes sense.